icu4j/main/classes/core/src/com/ibm/icu/text/RuleBasedBreakIterator.java

```diff
@@ -924,15 +925,12 @@ else if (mode == RBBI_RUN) {
                 }
 
                 //  Remember the break status (tag) values.
-                fRuleStatusIndex = stateTable[row + RBBIDataWrapper.TAGIDX];
-            }
-
-            int completedRule = stateTable[row + RBBIDataWrapper.ACCEPTING];
-            if (completedRule > 0 && completedRule != 0xffff) {
+                fRuleStatusIndex = stateTable[row + RBBIDataWrapper.TAGSIDX];
+            } else if (accepting >= 2) {
```

Comment:
same optional suggestion as in C++ -- test for 0 then 1=constant then just "else" ?